doc/hardware/porting/board_porting.rst

```diff
@@ -129,6 +129,14 @@ board directory in the zephyr repository, but it's the easiest way to get
 started. See :ref:`custom_board_definition` for documentation on moving your
 board directory to a separate repository once it's working.)
 
+.. note::
+
+  The board directory name does not need to match the name of the board.
+  Multiple boards can even defined be in one directory.
+  The most common example of this is for boards with multi-core SoC,
+  where each core is usually created as a logical board following the
+  naming scheme `<board>_<soc-core>`.
```

Comment:
Not sure I really like this. E.g. thingy53_nrf5340_cpuapp, thingy53_nrf5340_cpuapp_ns, thingy53_nrf5340_cpunet
The board is thingy53, the soc is nrf5340, the core, which uses an underscore, is cpuapp/cpunet, and there is optionally "_ns" for the application core for the non-secure domain. I don't think this text here really explains it, more so it makes it more confusing.

Reply:
I guess the point here is that a board dir may contain multiple board definitions.
Cause that itself is not always clear to users.

> The board is thingy53, the soc is nrf5340

Technically yes, but in Zephyr, the `thingy53_nrf5340` is considered the board in Zephyr context.
It's only because Nordic has decided to include the SoC in the name of the board.
Other boards may not do so, like the [mimxrt1170_evk](https://docs.zephyrproject.org/latest/boards/arm/mimxrt1170_evk/doc/index.html)

So the questions which remains are:
- How do we inform the users that a board dir may contain multiple board definitions
- Why and when is this possibility used ?
   For example users may do this for a board with a multi core SoC as described, but they should not start to put board `foo` and `bar` into a common board folder, even though that technically is possible.

@nordicjm do you have any proposal on how to provide this information to the reader in a short text, which doesn't have to deal with the fact that company A uses one naming convention, and company B another ?

Notice the text says: 
> where each core is **usually** created as a logical board following the naming scheme `<board>_<soc-core>`

so that means some boards may not do it like this or follow this naming scheme.

Because this text is short, and the main point is to inform users that a board dir may contain multiple board definitions with are small reasoning of why, then I'm giving +1 because I believe it's improving the existing documentation.

Reply:
Yeah the text here is just supposed to be an example and is supposed to show that the board directory name doesn't necessarily match the board name. But I suppose I could add a sentence clarifying this even further if it is a point of confusion still.